### PR TITLE
[fix][schema] Fix cherry-pick issue from #18283

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

#14608 Introduce a new break change about the delete schema logic of topic deletion starting from 2.11(milestone), #18283 cherry-picked this logic to branch-2.10 and branch-2.9. I'm wondering if it's better to roll back this kind of logic.
 
### Modifications

- Roll back delete schema logic of topic deletion.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
It should be a cherry-pick issue from https://github.com/apache/pulsar/pull/18283
It's better to change the PR title to "Fix cherry-pick issue from #18283"

**Review by yuruguo — APPROVED** [review]
LGTM

**Review by Technoboy- — APPROVED** [review]

**Review by yaalsn — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/PersistentTopicTest.java [hunk 1]
```diff
@@ -271,4 +272,70 @@ public void testPersistentPartitionedTopicUnload() throws Exception {
...
+        // after deleting the schema, try to create a topic with a different schema
+        try (Producer<Student> producer = pulsarClient.newProducer(Schema.AVRO(Student.class))
+                .topic(topicName)
+                .create()
+        ) {
+            Student student = new Student();
+            student.setName("Tom Jerry");
+            student.setAge(30);
+            student.setGpa(6);
+            student.setGpa(10);
```
Comment (yaalsn) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/PersistentTopicTest.java:320]: Is there any reason to invoke `setGpa` twice?

Comment (mattisonchao) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/PersistentTopicTest.java:319]: fixed

**Pull request merged**